cert/openssl.cnf

```diff
@@ -1,234 +1,424 @@
+countryName				= optional
+stateOrProvinceName		= optional
+localityName			= optional
+organizationName		= optional
 organizationalUnitName	= optional
-commonName		= supplied
-emailAddress		= optional
+commonName				= supplied
 
 ####################################################################
 [ req ]
-default_bits		= 2048
-default_keyfile 	= privkey.pem
-distinguished_name	= req_distinguished_name
-attributes		= req_attributes
-x509_extensions	= v3_ca	# The extentions to add to the self signed cert
+default_bits			= 2048
+distinguished_name		= req_distinguished_name
+x509_extensions			= v3_ca	
+string_mask 			= utf8only
+default_md				= sha384
+
+#req_extensions = v3_ca # The extensions to add to a certificate request
 
-# Passwords for private keys if not present they will be prompted for
-# input_password = secret
-# output_password = secret
+[ req_distinguished_name ]
 
-# This sets a mask for permitted string types. There are several options. 
-# utf8only: only UTF8Strings (PKIX recommendation after 2004).
-string_mask = utf8only
+#values
+countryName					= Country Name (2 letter code)
+stateOrProvinceName			= State or Province Name (full name)
+localityName				= Locality Name (eg, city)
+0.organizationName			= Organization Name (eg, company)
+organizationalUnitName		= Organizational Unit Name (eg, section)
+commonName					= Common Name
 
-req_extensions = sas_req # The extensions to add to a certificate request
+#min and max
+countryName_min				= 2
+countryName_max				= 2
+commonName_max				= 64
 
-[ req_distinguished_name ]
-countryName			= Country Name (2 letter code)
-countryName_default		= US
-countryName_min			= 2
-countryName_max			= 2
+#defaults
+countryName_default				= US
+stateOrProvinceName_default		= Washington, D.C.
+localityName_default 			= Washington, D.C.
+0.organizationName_default		= Wireless Innovation Forum
+organizationalUnitName_default	= www.wirelessinnovation.org
 
-stateOrProvinceName		= State or Province Name (full name)
+[ v3_ca ]
 
-localityName			= Locality Name (eg, city)
+subjectKeyIdentifier			= hash
+basicConstraints 				= critical, CA:true
+keyUsage 						= critical, digitalSignature, cRLSign, keyCertSign
 
-organizationName		= Organization Name (eg, company)
+####################################################################
+#Not used yet
+[ crl_section ]
 
-organizationalUnitName		= Organizational Unit Name (eg, section)
+URI.0 = http://crl3.digicert.com/WInnForumCA.crl
+URI.1 = http://crl4.digicert.com/WInnForumCA.crl
 
-commonName			= Common Name (domain name for SAS/domain proxy or applicant name)
-commonName_max			= 64
+####################################################################
+#Not used yet
+[ ocsp_section ] 
 
-emailAddress			= Email Address
-emailAddress_max		= 64
+caIssuers;URI.0 = http://cacerts.digicert.com/WinnForumCA.crt
+OCSP;URI.0 = http://ocsp.digicert.com
 
-[ req_attributes ]
+####################################################################
+[ wif_root_ca ]
```

Comment:
These are really great. The different sections are a great way to spell this out.